.github/workflows/python-tests.yml

```
@@ -0,0 +1,72 @@
+name: Python Tests
+
+# 1. Run on every push 
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    # 2. Use the latest Ubuntu runner
+    runs-on: ubuntu-latest
+
+    # 3. Spin up a PostgreSQL service for tests
+    services:
+      postgres:
+        image: postgres:15
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: test_db
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd "pg_isready -U postgres"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      # 4. Check out code
+      - uses: actions/checkout@v3
+
+      # 5. Wait for Postgres to be ready
+      - name: Wait for Postgres
+        run: |
+          until pg_isready -h localhost -p 5432 -U postgres; do
+            echo "Waiting for Postgres…"
+            sleep 1
+          done
+
+      # 6. Create test database & apply schema
+      - name: Initialize database
+        env:
+          PGPASSWORD: postgres
+        run: |
+          psql -h localhost -U postgres -c "DROP DATABASE IF EXISTS test_db;"
+          psql -h localhost -U postgres -c "CREATE DATABASE test_db;"
```

[postgres environment variables](https://www.postgresql.org/docs/current/libpq-envars.html) could simplify this a bit
```suggestion
        env:
          PGPASSWORD: postgres
          PGHOST: localhost
          PGUSER: postgres
        run: |
          psql -c "DROP DATABASE IF EXISTS test_db;"
          psql -c "CREATE DATABASE test_db;"
```